payment_processors/braintree_processor/models/payment_processors.py

```diff
@@ -94,18 +89,17 @@ def _update_transaction_status(self, transaction, result_transaction):
         :description: Updates a given transaction's data with data from a
                       braintreeSDK result payment method.
         """
-
         if not transaction.data:
             transaction.data = {}
 
-        transaction.data.update({
-            'status': result_transaction.status,
-            'braintree_id': result_transaction.id
-        })
+        transaction.external_reference = result_transaction.id
+        status = result_transaction.status
 
-        status = transaction.data['status']
+        transaction.data['status'] = status
```

Comment:
Because I used it a few lines below too, and didn't want to write `result_transaction.status` all the time.